### DISCUSSION TITLE
`@storybook/store` issue

### DISCUSSION THREAD

**Discussion by eugene-stativka** [percy/percy-storybook#480] [Support]
## `storyStoreV7` is not supported

`percy storybook ./storybook-static` fails with a message `Cannot call extract() unless you call cacheAllCSFFiles() first` if `storyStoreV7` feature flag is enabled.

## Environment

- Node version: 16.3.1
- `@percy/cli` version: 1.0.0-beta.71
- Version of Percy SDK you’re using:
- If needed, a build or snapshot ID:
- OS version: MacOS Monterey
- Type of shell command-line [interface]: zsh

**Comment by Robdel12** [comment]
👋🏼 Hey @eugene-stativka! Looks like this comment will be helpful: https://github.com/storybookjs/storybook/issues/13561#issuecomment-953724731

> In v7 mode, you need to await preview.cacheAllCSFFiles() before calling functions like store.getStoriesJson(). So we would need to do that in the extract script.

**Reply by eugene-stativka** [reply]
Hey @Robdel12! I thought this should happen inside the percy-storybook package. Am I right?

**Reply by Robdel12** [reply]
I wouldn't think so -- the Percy SDK takes the static output, loads it in a browser, grabs the list of stories, and then snapshots the story DOM. I would guess that this error is present if you tried to load the static-storybook build output in a browser (removing Percy from the flow).

**Reply by eugene-stativka** [reply]
This error happens only if I run `percy storybook ./storybook-static`. All other scenarios with Storybook work fine

**Reply by Robdel12** [reply]
Percy doesn't do anything to your storybook build though. We take the static build output and load it into the browser. The error sounds like it's an issue in the static output

**Comment by cooper667** [comment]
I had the same issue, although the thrown error is
```        
new Error(
          "Storybook object not found on the window. " +
            "Open Storybook and check the console for errors."
        )
```
which made it hard to track down!

Disabling `storyStoreV7` sorted it, but this will be the default in Storybook 7, so is worth looking into?

The static build works in Chrome, the only obvious difference is enabling `storyStoreV7` causes [this issue](https://github.com/storybookjs/storybook/issues/18424) which isn't present when doing a static build without

**Reply by Robdel12** [reply]
👋🏼 I'd give v4.3.0 a try: https://github.com/percy/percy-storybook/releases/tag/v4.3.0
